README.md

```diff
@@ -8,15 +8,112 @@
 
 ## Overview
 
-[//]: # (TODO: Add overview mentioning the purpose of the module, supported REST API versions, and other high-level details.)
+[Jira](https://www.atlassian.com/software/jira)  is a widely-used issue and project tracking software by Atlassian. It provides a REST API that allows applications to connect with Jira to create, update, and delete various resources, enabling efficient project and task management.
+
+The `ballerinax/jira` package offers APIs to interact with the [Jira REST API](https://developer.atlassian.com/cloud/jira/platform/rest/v3/intro), providing access to various endpoints for managing issues, filters, projects, and other resources.
 
 ## Setup guide
 
-[//]: # (TODO: Add detailed steps to obtain credentials and configure the module.)
+To use the Jira connector, you must have a Jira Cloud account with API access. If you do not have one, you can sign up for a Jira account on [Atlassian’s website](https://developer.atlassian.com/cloud/jira/platform/).
+
+### Step 1: Create an API Token
+1. Login to [Atlassian Account](https://developer.atlassian.com/cloud/jira/platform/)
+
+2. Click on the profile icon to get into developer console
+
+ <img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-jira/main/docs/setup/resources/atlassian-developer-console#1.png alt="Atlassian developer console" style="width: 70%;">
+
+3. In the new window, click on the profile again and go to 'Manage Account'
+
+4. Click and go to security tab and scroll down to the bottom to generate API token by clicking 'Create and Manage API token'
+
+ <img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-jira/main/docs/setup/resources/create-API-Token#2.png alt="Create API Token" style="width: 70%;">
+
+5. Give a name to the API token and copy and save it in a secret place
+
+<img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-jira/main/docs/setup/resources/generate-API-Token#3.png alt="Generate API Token" style="width: 70%;">
+
+### Step 2: Integrate JIRA for your account
+1. Click on the grid shaped icon in the right corner (Before Atlassian Account logo). Click on explore products
+
+<img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-jira/main/docs/setup/resources/try-jira-cloud#5.png alt="Try jira" style="width: 70%;">
```

Comment:
Rename the image in the resource folder.

Reply:
This image does not have the `Explore products` icon. It is there in the `try-jira-cloud` image. Could you please recheck if you have linked the correct images?